tests/test_lint_maturity.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+import os
+import sys
+from komodo import lint_maturity
+
+RELEASE_SAMPLE = """package_a1: v3.1.a1
+package_a2: 3.1.a2
+package_b1: v3.1.b1
+package_b2: 3.1.b1+ert
+package_rc1: v3.1.rc1
+package_rc2: 3.1.rc1
+package_NN1: v0.10.4
+package_NN2: 4.12.1+python2.7.14
+package_NN3: 1.8.6+pip
+package_NN4: 3.2.3.post2
+package_IV1: testing/2020.3/rc1
+package_IV2: 4.19.19-py3
+package_IV3: 5.13.1-src
+package_IV4: v1.2.3-nolibeclreq"""
+
+FILE_NAMES_SAMPLE = [
+    "2020.02.01-py27.yml",
+    "2020.02.a1-py27.yml",
+    "2020.02.b1-py27.yml",
+    "2020.02.rc1-py27.yml",
+    "bleeding-py27.yml",
+]
+
+EXPECTED_FOLDER = """2020.02.01-py27.yml has 10 packages with invalid maturity tag. 
+         Tag a packages: {'package_a1': 'v3.1.a1', 'package_a2': '3.1.a2'}
+         Tag b packages: {'package_b1': 'v3.1.b1', 'package_b2': '3.1.b1+ert'}
+         Tag rc packages: {'package_rc1': 'v3.1.rc1', 'package_rc2': '3.1.rc1'}
+         Tag Invalid packages: {'package_IV1': 'testing/2020.3/rc1', 'package_IV2': '4.19.19-py3', 'package_IV3': '5.13.1-src', 'package_IV4': 'v1.2.3-nolibeclreq'}
+2020.02.a1-py27.yml has 4 packages with invalid maturity tag. 
+         Tag Invalid packages: {'package_IV1': 'testing/2020.3/rc1', 'package_IV2': '4.19.19-py3', 'package_IV3': '5.13.1-src', 'package_IV4': 'v1.2.3-nolibeclreq'}
+2020.02.b1-py27.yml has 6 packages with invalid maturity tag. 
+         Tag a packages: {'package_a1': 'v3.1.a1', 'package_a2': '3.1.a2'}
+         Tag Invalid packages: {'package_IV1': 'testing/2020.3/rc1', 'package_IV2': '4.19.19-py3', 'package_IV3': '5.13.1-src', 'package_IV4': 'v1.2.3-nolibeclreq'}
+2020.02.rc1-py27.yml has 8 packages with invalid maturity tag. 
+         Tag a packages: {'package_a1': 'v3.1.a1', 'package_a2': '3.1.a2'}
+         Tag b packages: {'package_b1': 'v3.1.b1', 'package_b2': '3.1.b1+ert'}
+         Tag Invalid packages: {'package_IV1': 'testing/2020.3/rc1', 'package_IV2': '4.19.19-py3', 'package_IV3': '5.13.1-src', 'package_IV4': 'v1.2.3-nolibeclreq'}
+bleeding-py27.yml is incompatible with version naming 
+"""
+
+FOLDER_TEST = "/test_lint_maturity/"
+
+
+def create_test_files(RELEASE_SAMPLE, FILE_NAMES_SAMPLE, FOLDER_TEST):
+    folder_name = os.getcwd() + FOLDER_TEST
+    try:
+        os.mkdir(folder_name)
+    except:
+        sys.exit("Not possible to create folder with sample files.")
+
+    list_files = []
+    for file_name in FILE_NAMES_SAMPLE:
+        list_files.append(folder_name + file_name)
+        with open(folder_name + file_name, "w") as file_sample:
+            file_sample.write(RELEASE_SAMPLE)
+    return list_files
+
+
+def test_lint_maturity_folder():
+
+    list_files = create_test_files(RELEASE_SAMPLE, FILE_NAMES_SAMPLE, FOLDER_TEST)
+
+    with pytest.raises(SystemExit) as exit_info:
+        lint_maturity.run(list_files)
+
+    for file_name in list_files:
+        os.remove(file_name)
+    os.rmdir(os.path.dirname(list_files[0]))
```

Comment:
Is this for clean up? By running this in a `tmp` folder you wont have to worry about that, and it is one less place where your test can unexpectedly fail.